src/Vlingo.Common/Message/AsyncMessageQueue.cs

```diff
@@ -34,13 +36,15 @@ public AsyncMessageQueue(IMessageQueue deadLettersQueue)
             open = new AtomicBoolean(false);
             queue = new ConcurrentQueue<IMessage>();
 
-            executorThread = new Thread(ExecutorThreadStart);
-            executorThread.Start();
+            resetEvent = new AutoResetEvent(false);
+            cancellationSource = new CancellationTokenSource();
+            executorTask = new Task(TaskAction, cancellationSource.Token);
+            executorTask.Start();
```

Comment:
I don't have a context, but recommanded way would be to use `Task.Run`. If you construct a task using the task’s constructor, you then pay this synchronization cost when calling the Start method, because TPL need to protect against the chance that another thread is concurrently calling Start.

Reply:
Good catch. Updated the code.